### ISSUE TITLE
Error rendering doughnut chart with labels in Node.js

### ISSUE THREAD

**Issue by cesarlugoe** [chartjs/chartjs-plugin-datalabels#162]
Using 'chartjs-node-canvas'.

When trying to render a doughnut chart: 

```Promise {
  TypeError: Cannot read property 'x' of null
      at orient (/Users/cesar/hosco/chartjsIssue/chartjs-plugin-datalabels-doughnut/node_modules/chartjs-plugin-datalabels/dist/chartjs-plugin-datalabels.js:155:18)
 ............
}
```

- I can render a doughnut chart without activating `chartjs-plugin-datalabels`.
- I can render a Bar chart with `chartjs-plugin-datalabels` with the same setup.

repository:
https://github.com/cesarlugoe/chartjs-plugin-label-issue

**Comment by cesarlugoe** [comment]
I tried downgrading chart.js version to 2.7.3 from 2.8 as explained in this issue: https://github.com/chartjs/chartjs-plugin-datalabels/issues/150

But the error persists in my case.

**Comment by simonbrunel** [comment]
Hi @cesarlugoe! Sorry for the late reply. 

The following solution listed on that [page](https://github.com/SeanSobey/ChartjsNodeCanvas#loading-plugins) works for me:

```js
const chartJsFactory = () => {
    const Chart = require('chart.js');
    require('chartjs-plugin-datalabels');
    delete require.cache[require.resolve('chart.js')];
    delete require.cache[require.resolve('chartjs-plugin-datalabels')];
    return Chart;
}

// ...

const canvasRenderService = new CanvasRenderService(
  chartWidth,
  chartHeight,
  undefined,
  undefined,
  chartJsFactory
);
```

Since this issue is related to how `chartjs-node-canvas` import Chart.js (using fresh-require), let's track it in SeanSobey/ChartjsNodeCanvas#37.

**Comment by alvarouribe** [comment]
I fixed my issue.  (just register datalabels plugin LOCALLY per chart)

``` TypeError: Cannot read property 'x' of null```

ChartDataLabels plugin was registered globally in another chart that I had. instead I register locally and that fixed my issue. [Docs](https://chartjs-plugin-datalabels.netlify.app/guide/getting-started.html#registration)

basically I just had to remove this line in all my chart definitions 

```
// Register the plugin to all charts:
Chart.register(ChartDataLabels);
```

 instead I just need it the ChartDataLabels in my plugins object:
```var chart = new Chart(ctx, {
  plugins: [ChartDataLabels], ...```
